core/src/main/java/org/bitcoinj/core/TransactionWitness.java

```diff
@@ -41,6 +41,20 @@ public static TransactionWitness redeemP2WPKH(@Nullable TransactionSignature sig
         return witness;
     }
 
+    /**
+     * Creates the stack pushes necessary to redeem a P2WSH output.
+     */
+    public static TransactionWitness redeemP2WSH(byte[] witnessScript, TransactionSignature... signatures) {
```

Comment:
Would it maybe make sense to express the witnessScript as type Script?

Reply:
fixed